File: src/main/java/no/nav/tjenestepensjon/simulering/domain/TpLeverandor.java
package no.nav.tjenestepensjon.simulering.domain;

public class TpLeverandor {
    private final String name;
    private final String url;
    private final EndpointImpl impl;

    public TpLeverandor(String name, String url, EndpointImpl impl) {
        this.name = name;
        this.url = url;
        this.impl = impl;
    }

    public String getName() {
        return name;
    }

    public String getUrl() {
        return url;
    }

    public EndpointImpl getImpl() {
        return impl;
    }

    public enum EndpointImpl {
        SOAP,
        REST
    }
}

File: src/main/kotlin/no/nav/tjenestepensjon/simulering/model/domain/TpLeverandor.kt
package no.nav.tjenestepensjon.simulering.model.domain


data class TpLeverandor(val name: String,
                        val impl: EndpointImpl,
                        val simuleringUrl: String,
                        val stillingsprosentUrl: String) {

    enum class EndpointImpl {
        SOAP, REST
    }
}